apps/dapp-console-api/src/routes/contracts/ContractsRoute.ts

```diff
@@ -336,6 +340,46 @@ export class ContractsRoute extends Route {
       return { success: true }
     })
 
+  public readonly getContract = 'getContract' as const
+  public readonly getContractController = this.trpc.procedure
+    .use(isPrivyAuthed(this.trpc))
+    .input(
+      this.z.object({
+        contractId: this.z.string(),
+      }),
+    )
+    .query(async ({ ctx, input }) => {
+      const { contractId } = input
+      const { user } = ctx.session
+
+      if (!user) {
```

Comment:
we might want to throw this in a function so you don't need to rewrite in every endpoint something like `assertUserAuthenticated(user)`

Reply:
good call, added